textile/features.textile

```diff
@@ -54,8 +54,13 @@ h2(#test-guidelines). Test guidelines
 * @(G2)@ All tests by default are run against a special Ably sandbox environment.  This environment allows apps to be provisioned without any authentication that can then be used for client library testing. Bear in mind that all apps created in the sandbox environment are automatically deleted after 60 minutes and have low limits to prevent abuse. Apps are configured by sending a @POST@ request to @https://sandbox-rest.ably.io/apps@ with a JSON body that specifies the keys and their associated capabilities, channel namespace rules and any presence fixture data that is required; see "ably-common test-app-setup.json":https://github.com/ably/ably-common/blob/main/test-resources/test-app-setup.json. Presence fixture data is necessary for the REST library presence tests as there is no way to register presence on a channel in the REST library
 * @(G3)@ Testing statistics can be tricky due to timing issues and slow test suites as a result of sending requests to generate statistics.  As such, we provide a special stats endpoint in our sandbox environment that allows stats to be injected into our metrics system so that stats tests can make predictable assertions.  To create stats you must send an authenticated @POST@ request to the stats JSON to @https://sandbox-rest.ably.io/stats@ with the stats data you wish to create. See the "JavaScript stats fixture":https://github.com/ably/ably-js/blob/4e65d4e13eb8750a375b9511e4dd059092c0e481/spec/rest/stats.test.js#L8-L51 and "setup helper":https://github.com/ably/ably-js/blob/4e65d4e13eb8750a375b9511e4dd059092c0e481/spec/common/modules/testapp_manager.js#L158-L182 as an example
-* @(G4)@ This spec defines API version 1.2. A client library must identify to Ably the version of the spec it uses in all requests and connections, per "RSC7a":#RSC7a and "RTN2f":#RTN2f. The spec it uses is defined as the latest API version for which the library implements all spec items relating to the wire protocol
-** @(G4a)@ When sending the API version to Ably, the SDK must send the exact string specified in @G4@. Therefore, it is recommended that the SDK treat the version opaquely, as a string, not a float.
+* @(G4)@ This clause has been replaced by "@G5@":#G5 as of version 2.0 of this specification
+** @(G4a)@ This clause has been replaced by "@G5c@":#G5c as of version 2.0 of this specification
+* @(G5)@ A client library must identify the version of this specification it uses in all requests and connections to the Ably service, using the mechanisms described in "@RSC7e@":#RSC7e and "@RTN2h@":#RTN2h.
+** @(G5a)@ When sending the API version to Ably, the SDK must send an exact string in the form @MAJOR.MINOR@ where the @MAJOR@ and @MINOR@ components are defined according to "SemVer":https://semver.org/ in the "the @ably/specification@ repository":https://github.com/ably/specification at the commit containg the specification source to which the SDK is conforming.
```

Comment:
It's not clear from this what it means for the components to be defined "in the spec repo .. at the commit". Where in the repo? (It's clarified by the "Versioning" paragraph in the readme (which says: in package.json), but that's a bit hidden, someone reading the spec won't necessarily know to look there)

Reply:
I'll admit I had several runs at getting this sentence "right", yet clearly didn't... I'll have a re-think on how to articulate this. Though, perhaps, my [other response](https://github.com/ably/specification/pull/105#issuecomment-1286029865) might have helped you to be able to assist me in re-writing this...?

Reply:
Re-reading this conversation in the light of a new day, I'm not sure I can think of how to re-write this any more clearly. The guiding principle for this work is [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), hence why I remain of the belief that once a reader of this spec point navigates to the source repository that they will find the "Versioning" header+paragraph added in https://github.com/ably/specification/pull/105/commits/860575c43af92bbe884c3cdfafa524b0e1c86755, so this detail does not need repeating elsewhere. I don't think that needs stronger clarification. And, if in future, we decide to move the version definition from `package.json` to some other location then there's only one place we need to update instructions in this repository (the root readme). Also, [as I've indicated elsewhere](https://github.com/ably/specification/pull/105#issuecomment-1286029865), I don't feel the version number is "data" (i.e. spec point worthy), it's "meta data" relating to the source code base as a whole.

Reply:
I think it is simpler if the spec explicitly states the version number it represents, rather than stating the commit it is from with the reader then needing to lookup what version that commit corresponds to.

Perhaps we should be considering templating this file, so in the source it has something like:

```
This spec defines API version {{ VERSION }}.
```

and then we render it so that the published spec says:

```
This spec defines API version 2.0.
```

Reply:
I think that if @lmars's suggested approach works, then that would preferable.